tests/conftest.py

```diff
@@ -188,6 +196,36 @@ def robot_service_thread(container: ApplicationContainer):
     robot_service_thread.join()
 
 
+@pytest.fixture
+def mocked_robot_service(container: ApplicationContainer, mocker):
+    robot_service: Robot = Robot(
+        events=container.events(),
+        robot=container.robot_interface(),
+        shared_state=container.shared_state(),
+        mqtt_publisher=container.mqtt_client(),
+    )
+
+    mocker.patch.object(RobotStartMissionThread, "run", return_value=lambda: None)
+    mocker.patch.object(RobotBatteryThread, "run", return_value=lambda: None)
+    mocker.patch.object(RobotStatusThread, "run", return_value=lambda: None)
+    mocker.patch.object(RobotMonitorMissionThread, "run", return_value=lambda: None)
+    mocker.patch.object(RobotStopMissionThread, "run", return_value=lambda: None)
+    mocker.patch.object(RobotPauseMissionThread, "run", return_value=lambda: None)
+    mocker.patch.object(RobotResumeMissionThread, "run", return_value=lambda: None)
+    mocker.patch.object(RobotUploadInspectionThread, "run", return_value=lambda: None)
+
+    mocker.patch.object(RobotStartMissionThread, "join", return_value=lambda: None)
+    mocker.patch.object(RobotBatteryThread, "join", return_value=lambda: None)
+    mocker.patch.object(RobotStatusThread, "join", return_value=lambda: None)
+    mocker.patch.object(RobotMonitorMissionThread, "join", return_value=lambda: None)
+    mocker.patch.object(RobotStopMissionThread, "join", return_value=lambda: None)
+    mocker.patch.object(RobotPauseMissionThread, "join", return_value=lambda: None)
+    mocker.patch.object(RobotResumeMissionThread, "join", return_value=lambda: None)
+    mocker.patch.object(RobotUploadInspectionThread, "join", return_value=lambda: None)
```

Comment:
Technically this could be done in each test, but it's convenient to not have to unless it's relevant.

Reply:
And obviously if it was done in each test you wouldn't need to do it for all the threads. But this code prevents the threads from actually running.